### PR TITLE
feat(unlock-protocol-com): Add "State of Unlock" page

### PR DESCRIPTION
**Pull request by m00n620**
# Description

Implement State page on unlockprotocol.com

# Screenshots

### CODE REVIEW

File: unlock-protocol-com/package.json [hunk 1]
```diff
@@ -4,6 +4,8 @@
   "private": true,
   "dependencies": {
     "@babel/core": "7.20.2",
+    "@ethersproject/abi": "^5.7.0",
+    "@ethersproject/providers": "^5.7.2",
```
Comment (searchableguy) [inline unlock-protocol-com/package.json:8]: Pin dependency versions. 👍 

There are few other in the same file.

File: unlock-protocol-com/src/components/icons/GrossNetworkIcons.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            fillRule="evenodd"
+            clipRule="evenodd"
+            d="M20.8999 24.6286C24.7816 24.6286 27.9284 21.4818 27.9284 17.6C27.9284 13.7182 24.7816 10.5714 20.8999 10.5714C17.0181 10.5714 13.8713 13.7182 13.8713 17.6C13.8713 21.4818 17.0181 24.6286 20.8999 24.6286ZM20.8999 27.2C26.2018 27.2 30.4999 22.9019 30.4999 17.6C30.4999 12.2981 26.2018 8 20.8999 8C15.5979 8 11.2999 12.2981 11.2999 17.6C11.2999 22.9019 15.5979 27.2 20.8999 27.2Z"
+            fill="#35D07F"
+          />
+        </g>
+      </svg>
+    </IconBase>
+  )
+}
```
Comment (searchableguy) [inline unlock-protocol-com/src/components/icons/GrossNetworkIcons.tsx:191]: We actually import crypto-icon package in unlock-app for exactly this. That supports all tokens, not just these. 

Can you please add it to the site and use it here? That would be save the boilerplate code.

File: unlock-protocol-com/src/components/pages/State/index.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+        const subgraphData = await querySubgraph(
+          'https://api.studio.thegraph.com/query/37457/test-unlock/v0.0.8/',
+          subgraphConfig
+        )
+        setSubgraphData(subgraphData.data)
+        console.log(subgraphData)
+      }
+    }
+    run()
+  }, [])
```
Comment (searchableguy) [inline unlock-protocol-com/src/components/pages/State/index.tsx:233]: use react-query if possible. That would cache and re-fetch in an interval on its own.

File: unlock-protocol-com/src/components/pages/State/index.tsx [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+        .map((item) => ({
+          ...item,
+          Icon: GROSS_NETWORK_ICONS.find(
+            (icon) =>
+              icon.unit.toLowerCase() ===
+              item.network.baseCurrencySymbol.toLowerCase()
+          )?.Icon,
+        }))
+        .filter((item) => !item.network.isTestNetwork)
+      console.log(gnpValueswithIcon)
```
Comment (searchableguy) [inline unlock-protocol-com/src/components/pages/State/index.tsx:253]: remove?

**Comment by ChialiTsai** [comment]
Hi, is there a way that I can view the live version, would you like to give feedback on the UI and the responsiveness, thanks!

File: unlock-protocol-com/package.json (hunk 1 above)
Comment (m00n620) [reply unlock-protocol-com/package.json]: resolved

File: unlock-protocol-com/src/components/icons/GrossNetworkIcons.tsx (hunk 2 above)
Comment (m00n620) [reply unlock-protocol-com/src/components/icons/GrossNetworkIcons.tsx]: resolved

File: unlock-protocol-com/src/components/pages/State/index.tsx (hunk 3 above)
Comment (m00n620) [reply unlock-protocol-com/src/components/pages/State/index.tsx]: resolved

File: unlock-protocol-com/src/components/pages/State/index.tsx (hunk 4 above)
Comment (m00n620) [reply unlock-protocol-com/src/components/pages/State/index.tsx]: resolved

**Comment by m00n620** [comment]
> Hi, is there a way that I can view the live version, would you like to give feedback on the UI and the responsiveness, thanks!

Can you please run it on your local?

**Comment by julien51** [comment]
Please merge conflicts and let's do this!

**Comment by m00n620** [comment]
> Please merge conflicts and let's do this!

Resolved the conflicts, @julien51

**Comment by m00n620** [comment]
Resolved the conflicts, @julien51

File: unlock-protocol-com/package.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
     "@unlock-protocol/ui": "workspace:./packages/ui",
     "@unlock-protocol/unlock-assets": "workspace:./packages/unlock-assets",
     "@zeit/next-source-maps": "0.0.3",
+    "apexcharts": "^3.36.0",
```
Comment (julien51) [inline unlock-protocol-com/package.json:22]: please pin dependencies

**Comment by julien51** [comment]
You are missing `react-crypto-icons` in the dependencies.

File: unlock-protocol-com/src/components/icons/ActiveLock.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { IconBase, IconBaseProps } from 'react-icons'
+
+export function ActiveLock(props: IconBaseProps) {
```
Comment (julien51) [inline unlock-protocol-com/src/components/icons/ActiveLock.tsx:4]: Please move to `unlock-assets` package

File: unlock-protocol-com/src/components/icons/Key.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { IconBase, IconBaseProps } from 'react-icons'
+
+export function Key(props: IconBaseProps) {
```
Comment (julien51) [inline unlock-protocol-com/src/components/icons/Key.tsx:4]: Please move to `unlock-assets` package

File: unlock-protocol-com/src/components/icons/Lock.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { IconBase, IconBaseProps } from 'react-icons'
+
+export function Lock(props: IconBaseProps) {
```
Comment (julien51) [inline unlock-protocol-com/src/components/icons/Lock.tsx:4]: Please move to `unlock-assets` package

File: unlock-protocol-com/src/components/interface/Navigation/index.tsx [hunk 9]
```diff
@@ -342,6 +342,9 @@ function Desktop() {
       <Link className={link} href="https://unlock-protocol.com/guides">
         Creators
       </Link>
+      <Link className={link} href="/state">
```
Comment (julien51) [inline unlock-protocol-com/src/components/interface/Navigation/index.tsx:345]: Let's not add it here yet.

File: unlock-protocol-com/src/components/pages/State/index.tsx [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+  },
+  {
+    value: 281,
+    title: 'Active Locks',
+    description: 'Minted at least 1 membership in the last 30 days',
+    Icon: ActiveLock,
+  },
+]
+
+const NETWORKS = [
```
Comment (julien51) [inline unlock-protocol-com/src/components/pages/State/index.tsx:46]: Please use our package networks

Comment (julien51) [reply unlock-protocol-com/src/components/pages/State/index.tsx:46]: And filter any network for which `isTest` is true.

File: unlock-protocol-com/src/utils/apiRequest.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ethers } from 'ethers'
+import { networks } from '@unlock-protocol/networks'
+
+async function getGdpForNetwork(provider, network) {
```
Comment (julien51) [inline unlock-protocol-com/src/utils/apiRequest.ts:5]: No need to do that since we are getting this from the subgraph,, Correct?

**Review by julien51 — CHANGES_REQUESTED** [review]
Good start! Let's hook up the real data now that we have it!

**Comment by m00n620** [comment]
@julien51, resolved all your comments and it shows real data now!

**Comment by julien51** [comment]
Let me look at this! This is very cool!

**Comment by julien51** [comment]
Screen Shot 2022-11-18 at 6 29 32 PM
I think tis is missing a bunch of things, especially on the Polygon front

File: unlock-protocol-com/src/config/routes.tsx [hunk 12]
```diff
@@ -92,4 +92,12 @@ export const routes: Routes = {
         'Join the Unlock Protocol community and core teams for conferences, workshops, AMA, Governance and, other activities',
     }),
   },
+  state: {
+    label: 'State of Unlock',
+    seo: customizeSEO({
+      path: '/state',
+      title: 'State of Unlock',
+      description: 'State of Unlock',
```
Comment (julien51) [inline unlock-protocol-com/src/config/routes.tsx:100]:
```suggestion
      description: 'The state of the Unlock Protocol network, with total locks deployed, active locks and total keys minted. It displays data across all support blockchains.',
```

File: unlock-protocol-com/src/utils/apiRequest.ts (hunk 11 above)
Comment (julien51) [reply unlock-protocol-com/src/utils/apiRequest.ts:5]: @m00n620 please confirm and delete if not needed!

File: unlock-protocol-com/src/components/pages/State/index.tsx [hunk 13]
```diff
@@ -0,0 +1,567 @@
...
+          break
+        }
+      }
+
+      setXaxis({
+        categories: xAxisLabels,
+      })
+      setSeries([
+        {
+          name: 'Keys Sold',
```
Comment (julien51) [inline unlock-protocol-com/src/components/pages/State/index.tsx:298]:
```suggestion
          name: 'Keys (Memberships) Minted',
```

**Comment by julien51** [comment]
We're missing some important elements in the subgraph https://github.com/unlock-protocol/unlock/pull/10737

**Comment by julien51** [comment]
localhost_3002_blog_state-of-unlock

**Comment by julien51** [comment]
closing.

**Pull request closed without merging**